### DISCUSSION TITLE
Search not taking "display name" of the part into account

### DISCUSSION THREAD

**Discussion by turist79** [inventree/InvenTree#8577] [Q&A] [answered]
Hi

searching for stock items or parts does not take revision/ipn (or anything else) concatenated to the part name into account, searches only on the actual fields:

![image](https://github.com/user-attachments/assets/44dd88fb-7207-4169-bafc-363122c94ed5)

note "21006909-000" as part of concatenated part name.
searching for "6909" returns nothing

Possible solution: add all the part fields to the field selector for parts/stock items lists and all the stock item fields to the stock items list. As an additional for the stock items - add manufacturer part details in the same manner as supplier part, currently there is no way of getting info on the manufacturer of the item except going into itemdetails.

Question: is this something plugin engine is capable of doing? I looked through all the plugin examples and documentation but did not found hints on how to modify the existing lists (adding columns currently not in the selection list, adding custom filters, change display names (for example supplier part is being displayed as SKU and i would like having Sulliper/SKU). Any hint would be helpful.

**Comment by matmair** [comment]
Plugins can not interact with the search system; that is too low level.

**Comment by SchrodingersGat** [comment, marked as answer]
The "full name" is not stored anywhere in the database - it is constructed "on demand". As such, we cannot "search" for this field. We would probably have to come up with some db storage solution for this to work
